apps/server/src/modules/tldraw/service/tldraw.ws.service.spec.ts

```diff
@@ -0,0 +1,376 @@
+import { Test } from '@nestjs/testing';
+import { INestApplication } from '@nestjs/common';
+import WebSocket from 'ws';
+import { WsAdapter } from '@nestjs/platform-ws';
+import { CoreModule } from '@src/core';
+import { ConfigModule } from '@nestjs/config';
+import { createConfigModuleOptions } from '@src/config';
+import { config } from '@src/modules/tldraw/config';
+import { WsSharedDocDo } from '@src/modules/tldraw/types';
+import { TextEncoder } from 'util';
+import * as SyncProtocols from 'y-protocols/sync';
+import * as AwarenessProtocol from 'y-protocols/awareness';
+import { encoding } from 'lib0';
+import { TldrawBoardRepo } from '@src/modules/tldraw/repo';
+import { TldrawWs } from '@src/modules/tldraw/controller';
+import { TldrawWsService } from '.';
+import { TestHelper } from '../helper/test-helper';
+
+describe('TldrawWS', () => {
+	let app: INestApplication;
+	let ws: WebSocket;
+	let service: TldrawWsService;
+
+	const gatewayPort = 3346;
+	const wsUrl = TestHelper.getWsUrl(gatewayPort);
+	const testMessage =
```

Comment:
is it important that the message is the same in every testcase? if not, I would rather have it movend into the setup functions. You can create a shared setup function used by the other setup functions to avoid duplicated code